api/views/event.py

```diff
@@ -28,3 +29,27 @@ def get_event_types(self, request, pk=None):
         Returns a dictionary of the defined event types
         """
         return Response(EVENT_TYPE_CHOICES)
+
+    @action(detail=False, methods=['post'], url_path="duplicate-event")
+    def duplicate_event(self, request):
+        """
+        Duplicates an event as well as the questions within the event.
+        """
+        event = Event.objects.filter(id=request.data.get("event", None)).first()
```

Comment:
We have to check if they exist first and respond accordingly with 404 and the correct message if they don't.

Reply:
I think this is not resolved.

Reply:
Kinda confused why. If I make a request with an invalid course or event through postman, I get a 404 error with the defined error message.
![Screen Shot 2021-07-22 at 3 30 12 PM](https://user-images.githubusercontent.com/59884796/126698578-9f9dd059-138d-47ee-9c5a-9fd365331671.png)

Reply:
Ok yes, I see you did the validation at the end of the function. I think it should be at the beginning of the function. You can use `get_object_or_404` shortcut.

Reply:
Okay yeah that makes sense!